### PR TITLE
[3.10] Improve core update quickicon as well as none-supported update messages

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for a first subset of the issues reported here #36610 .

### Summary of Changes

- Make sure we use the current information we got from the detailsurl on the `getUpdateInformation` method
- Check whether we have got actually an update from the detailsurl or not.
- Improve joomla update quickicon to use `getUpdateInformation` vs something from com_installer => This should improve the issues with that quickicon pointing to an outdated version too once merged up to J4
- Deprecate `$this->updateInformation['current']` and move the core to use `$this->updateInformation['installed']` => Only keep that old value so that possible admin template overrides are still working

## First test
- Install 3.10.6-rc1 on php 5.6 (or on any other none supported setup for J4)
- switch to the testing update server
- apply this patch
- Check the Joomla update quickicon (=> The update should not be shown)
- go to Joomla Update (=> You only see reinstall)

## seccond test
- Install 3.10.6-rc1 on a J4 supported setup 
- switch to the testing update server
- apply this patch
- Check the Joomla update Quickicon (=> Update to J4 are shown)
- go to Joomla Update (=> Update to J4 are shown)

### Actual result BEFORE applying this Pull Request

image

### Expected result AFTER applying this Pull Request

image

### Documentation Changes Required

none

### CODE REVIEW

File: administrator/components/com_joomlaupdate/models/default.php [hunk 1]
```diff
@@ -223,7 +215,15 @@ public function getUpdateInformation()
 		$update = new JUpdate;
 		$update->loadFromXML($updateObject->detailsurl, $minimumStability);
 
+		// Make sure we use the current information we got from the detailsurl
 		$this->updateInformation['object'] = $update;
+		$this->updateInformation['latest'] = $update->get('version')->_data;
+
+		// Check whether we have got actually an update from the detailsurl or not.
```
Comment (brianteeman) [inline administrator/components/com_joomlaupdate/models/default.php]:
```suggestion
		// Check whether we have got an update from the detailsurl or not.
```

File: media/plg_quickicon_joomlaupdate/js/jupdatecheck.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ jQuery(document).ready(function() {
 			// An error occurred
 			jQuery('#plg_quickicon_joomlaupdate').find('span.j-links-link').html(plg_quickicon_joomlaupdate_text.ERROR);
 		},
-		url: plg_quickicon_joomlaupdate_ajax_url + '&eid=700&cache_timeout=3600'
+		url: plg_quickicon_joomlaupdate_ajax_url
```
Comment (brianteeman) [inline media/plg_quickicon_joomlaupdate/js/jupdatecheck.js:50]: this looks wrong

Comment (zero-24) [reply media/plg_quickicon_joomlaupdate/js/jupdatecheck.js:50]: The ajax URL is now pointing to a new com_joomlaupdate ajax endpoint which does not need that parameters anymore. ;)

Comment (brianteeman) [reply media/plg_quickicon_joomlaupdate/js/jupdatecheck.js:50]: ok

File: plugins/quickicon/joomlaupdate/joomlaupdate.php [hunk 3]
```diff
@@ -38,21 +38,20 @@ class PlgQuickiconJoomlaupdate extends JPlugin
...
 
-		$cur_template = JFactory::getApplication()->getTemplate();
+		$currentTemplate = JFactory::getApplication()->getTemplate();
 
-		$token    = JSession::getFormToken() . '=' . 1;
-		$url = JUri::base() . 'index.php?option=com_joomlaupdate';
-		$ajax_url = JUri::base() . 'index.php?option=com_installer&view=update&task=update.ajax&' . $token;
-		$script = array();
+		$url      = JUri::base() . 'index.php?option=com_joomlaupdate';
+		$ajaxUrl  = JUri::base() . 'index.php?option=com_joomlaupdate&task=update.ajax&' . JSession::getFormToken() . '=' . 1;
```
Comment (Quy) [inline plugins/quickicon/joomlaupdate/joomlaupdate.php]:
```suggestion
		$ajaxUrl  = JUri::base() . 'index.php?option=com_joomlaupdate&task=update.ajax&' . JSession::getFormToken() . '=1';
```

**Comment by pritam825** [comment]
it's almost same as expected, can you please review my screenshot so that I will consider as an successful testing, thanks in advance.
Screenshot (2010)

**Comment by zero-24** [comment]
> it's almost same as expected, can you please review my screenshot so that I will consider as an successful testing, thanks in advance.

@pritam825 I'm sorry I did not saw your message earlier yes thats a good test you can try to hit "Check for Updates" to get the reinstall page again too.

**Comment by pritam825** [comment]
@zero-24, no worry 😅
   thanks 😊

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on c7020925c20f5168d069cc62011d5d127aa7d7a0<br /><br />Thanks @zero-24 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36950">issues.joomla.org/tracker/joomla-cms/36950</a>.</sub>

File: administrator/components/com_joomlaupdate/models/default.php [hunk 4]
```diff
@@ -223,7 +215,15 @@ public function getUpdateInformation()
 		$update = new JUpdate;
 		$update->loadFromXML($updateObject->detailsurl, $minimumStability);
 
+		// Make sure we use the current information we got from the detailsurl
 		$this->updateInformation['object'] = $update;
+		$this->updateInformation['latest'] = $update->get('version')->_data;
```
Comment (zero-24) [inline administrator/components/com_joomlaupdate/models/default.php]:
```suggestion
		$this->updateInformation['latest'] = $updateObject->version;
```

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on eeb5aed8c34ec495612746e80d7152251b63b022<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36950">issues.joomla.org/tracker/joomla-cms/36950</a>.</sub>

Bildschirmfoto 2022-06-11 um 18 55 03

**Comment by zero-24** [comment]
Thanks @tecpromotion  and @SniperSister for finding the longstanding issue with the strtolower which caused this mess in the first place. Yes it was me who did the bad code :)

**Pull request merged**